GRPC miner migration v2 `/_sign` Handler

twiny opened the pull request.

twiny committed:
generated Sign classes and method

twiny committed:
added Sign messages and service

twiny committed:
added Sign method to minerGRPCService struct

twiny committed:
embedded minerproto.UnimplementedMinerServiceServer to minerGRPCService struct

twiny committed:
added Sign grpc handler

kushthedude commented on code/go/0chain.net/miner/server/grpc/handlers/sign.go:
```diff
@@ -0,0 +1,40 @@
+	privateKey := req.PrivateKey
+	publicKey := req.PublicKey
+	data := req.Data
+	timestamp := req.TimeStamp
```
They are unnecessary assignment we can directly use `req.*` variable whereever need be, HTTP Handler has such because it has to extract from the http.Request

kushthedude commented on code/go/0chain.net/miner/server/grpc/handlers/sign.go:
```diff
@@ -0,0 +1,40 @@
+	timestamp := req.TimeStamp
+	key, err := hex.DecodeString(publicKey)
+	if err != nil {
+		return nil, err
```
use error wrapping 
```suggestion
		return nil, errors.New(err, "failed to decode string")
```

kushthedude commented on code/go/0chain.net/miner/server/grpc/handlers/sign.go:
```diff
@@ -0,0 +1,40 @@
+	hash := encryption.Hash(hashdata)
+	signature, err := encryption.Sign(privateKey, hash)
+	if err != nil {
+		return nil, err
```
Use error wrapping

kushthedude reviewed (commented):
Tests are missing for the PR, would you mind adding some unit tests or integration tests for the handler?

twiny committed:
removed unnecessary variable assignment

twiny committed:
added client/server test & sign method test

twiny committed:
used pkg "google.golang.org/grpc/test/bufconn" to conduct test
and add some test cases.

twiny committed:
todo

twiny committed:
Merge branch 'grpc-migration-v2' into grpc-migration-v2-dev.00

twiny committed:
go mod tidy

twiny committed:
added UnimplementedMinerServiceServer field initialization

twiny committed:
using assert to conduct tests
removed grpc server/client connection
used directly UnimplementedMinerServiceServer

twiny committed:
removed unnecessary code

twiny committed:
fixup! removed unnecessary code

twiny committed:
moved sign_test.go file to /handler dir. updated test cases

twiny committed:
removed unused field in test struct
